tools/flowc/typechecker/combine_types.flow

```diff
@@ -292,23 +292,27 @@ combinePositiveAndNegative(env : FcTypeEnv, positive : FcType, negative : FcType
 				switch (negative) {
 					FcTypeStruct(ns, stp, sargs, si): {
 						if (ps == ns && length(ptp) == length(stp)) {
-							[FcTypeStruct(
-								ns,
-								mapi(ptp, \i, pt -> {
-									if (i < length(stp)) {
-										ct = combinePositiveAndNegative(env, pt, stp[i], onError);
-										if (ct != []) ct[0] else pt
-									} else pt
-								}),
-								mapi(pargs, \i, parg -> {
-									FcStructArg(
-										parg.name,
-										combinePositiveAndNegative(env, parg.type, sargs[i].type, onError)[0],
-										parg.ismutable
-									)
-								}),
-								si
-							)]
+							pargs_combined = mapi(pargs, \i, parg -> {
```

Comment:
filtermapi, and then check the length instead of first map, then forall, then filtermap.